### PR TITLE
Expired token bug ver2

### PR DESCRIPTION
**Pull request by medegor44**

### CODE REVIEW

File: src/hooks/useAuthorization.js [hunk 1]
```diff
@@ -10,23 +10,34 @@ const useAuthorization = () => {
...
+        setUserData(null);
+        clearToken();
         return;
       }
+
+      setAuthorized(true);
       setUserData({ token, ...user });
     };
 
+    const redirectToOrigin = () => {
```
Comment (medegor44) [inline src/hooks/useAuthorization.js]: Maybe I need to remove redirectToOrigin from current PR and open another one for this improvement.

Comment (confirmit-horizons) [reply src/hooks/useAuthorization.js]: Making a redirect inside a hook is bad idea. If you really need to perform a redirect then you should do it in a component based on info returned from the hook.
Why do you need a redirect?

File: src/hooks/useAuthorization.js [hunk 2]
```diff
@@ -10,23 +10,34 @@ const useAuthorization = () => {
...
 
       const token = getToken();
-      setAuthorized(true);
 
       const user = await fetchUser(`${token}`);
 
       if (user.hasError) {
         setAuthorized(false);
+        setUserData(null);
+        clearToken();
```
Comment (confirmit-horizons) [inline src/hooks/useAuthorization.js]: Better solution would be to clear the token on any API request failure and handle empty token case in a service (for example, show an error and let user to re-login again)

File: src/hooks/useAuthorization.js (hunk 1 above)
Comment (medegor44) [reply src/hooks/useAuthorization.js]: I need redirect to clear token info from url.

**Pull request closed without merging**